### PR TITLE
expression: Disallow conv fuction with hybrid type argement to be pushed down to TiKV

### LINKED ISSUE #51877: Tikv conv function inconsistent results with tidb
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
set tidb_opt_projection_push_down = 'on';
create table bug15583(b BIT(8), n INT);
insert into bug15583 values(128, 128);
insert into bug15583 values(null, null);
insert into bug15583 values(0, 0);
insert into bug15583 values(255, 255);
select conv(b, 10, 2), conv(b + 0, 10, 2) from bug15583;

### 2. What did you expect to see? (Required)
conv(b, 10, 2)  conv(b + 0, 10, 2)
10000000       10000000
NULL    NULL
0       0
11111111       11111111

### 3. What did you see instead (Required)
conv(b, 10, 2)  conv(b + 0, 10, 2)
0       10000000
NULL    NULL
0       0
0       11111111

### 4. What is your TiDB version? (Required)

| Release Version: v7.4.0-alpha-2012-gfe01d11df6-dirty
Edition: Community
Git Commit Hash: https://github.com/pingcap/tidb/commit/fe01d11df67ca60573b028ed4c9bc52db9a17a7e
Git Branch: div_increase_prec
UTC Build Time: 2024-03-15 08:10:07

### PR DESCRIPTION
**Pull request by yibin87**
### What problem does this PR solve?

Issue Number: close #51877

Problem Summary:

### What changed and how does it work?
As described in #51877, such case can't be supported in TiKV side, just disallow it to be pushed down.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/expression/expr_to_pb_test.go
Comment (xzhangxian1008) [inline pkg/expression/expr_to_pb_test.go:1]: Maybe we can test pushdown with integration test like [this](https://github.com/pingcap/tidb/pull/46329/files#diff-fa42dbc5f88d5aef4ee1cb7452ee4497fc545324b76bee0e44f48e5f1af9f512R5285)

Comment (yibin87) [reply pkg/expression/expr_to_pb_test.go:1]: Nice suggestion. For this case, the push down check is very simple and straightforward, I think the expr_to_pb_test can cover the functionality alread.

File: pkg/expression/infer_pushdown.go [hunk 1]
```diff
@@ -221,6 +221,17 @@ func scalarExprSupportedByTiKV(sf *ScalarFunction) bool {
...
 		ast.UUID:
 
+		return true
+	// Rust use the llvm math functions, which have different precision with Golang/MySQL(cmath)
+	// open the following switchers if we implement them in coprocessor via `cmath`
+	case ast.Conv:
+		arg0 := sf.GetArgs()[0]
+		// To be aligned with MySQL, tidb handles hybrid type argument specially, tikv can't be consistent with tidb now.
+		if f, ok := arg0.(*ScalarFunction); ok {
+			if f.FuncName.L == ast.Cast && f.GetArgs()[0].GetType().Hybrid() {
```
Comment (windtalker) [inline pkg/expression/infer_pushdown.go:231]: Don't need to check `IsBinaryLiteral(f.GetArgs()[0])` here?

Comment (yibin87) [reply pkg/expression/infer_pushdown.go]: Not quite sure if constant folding may not take effect. Added for safety.

**Review by xzhangxian1008 — APPROVED** [review]
LGTM

**Comment by zanmato1984** [comment]
Please fill in proper release notes message next time @yibin87 , thanks.

**Pull request merged**